dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfIntegrationTest.scala

```diff
@@ -0,0 +1,189 @@
+import org.bitcoins.core.crypto.{
+  ECPrivateKey,
+  Schnorr,
+  SchnorrNonce,
+  Sha256DigestBE
+}
+import org.bitcoins.core.currency.{Bitcoins, CurrencyUnits}
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.script.P2PKHScriptPubKey
+import org.bitcoins.core.protocol.transaction.TransactionOutPoint
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.util.CryptoUtil
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+import org.bitcoins.core.wallet.utxo.P2PKHSpendingInfo
+import org.bitcoins.testkit.rpc.BitcoindRpcTestUtil
+import org.bitcoins.testkit.util.BitcoindRpcTest
+import org.scalatest.Assertion
+import scodec.bits.ByteVector
+
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+class BinaryOutcomeDLCWithSelfIntegrationTest extends BitcoindRpcTest {
+  private val clientsF = BitcoindRpcTestUtil.createNodePairV18(clientAccum)
+  private val clientF = clientsF.map(_._1)
+
+  behavior of "BinaryOutcomeDLCWithSelf"
+
+  it should "be able to publish all DLC txs to Regtest and have them accepted for all cases" in {
+    val outcomeWin = "WIN"
+    val outcomeWinHash =
+      CryptoUtil.sha256(ByteVector(outcomeWin.getBytes)).flip
+    val outcomeLose = "LOSE"
+    val outcomeLoseHash =
+      CryptoUtil.sha256(ByteVector(outcomeLose.getBytes)).flip
+    val oraclePrivKey = ECPrivateKey.freshPrivateKey
+    val oraclePubKey = oraclePrivKey.publicKey
+    val preCommittedK = SchnorrNonce.freshNonce
+    val preCommittedR = preCommittedK.publicKey
+    val localInput = CurrencyUnits.oneBTC
+    val remoteInput = CurrencyUnits.oneBTC
+
+    val inputPrivKeyLocal = ECPrivateKey.freshPrivateKey
+    val inputPubKeyLocal = inputPrivKeyLocal.publicKey
+    val inputPrivKeyRemote = ECPrivateKey.freshPrivateKey
+    val inputPubKeyRemote = inputPrivKeyRemote.publicKey
+
+    val localAddress =
+      BitcoinAddress.fromScriptPubKey(P2PKHScriptPubKey(inputPubKeyLocal),
+                                      RegTest)
+    val remoteAddress =
+      BitcoinAddress.fromScriptPubKey(P2PKHScriptPubKey(inputPubKeyRemote),
+                                      RegTest)
+
+    val changePrivKey = ECPrivateKey.freshPrivateKey
+    val changePubKey = changePrivKey.publicKey
+    val changeSPK = P2PKHScriptPubKey(changePubKey)
+
+    def executeForCase(outcomeHash: Sha256DigestBE,
+                       local: Boolean): Future[Assertion] = {
+      val oracleSig =
+        Schnorr.signWithNonce(outcomeHash.bytes, oraclePrivKey, preCommittedK)
+
+      val fundedInputsTxidF = for {
+        client <- clientF
+        transactionWithoutFunds <- client
+          .createRawTransaction(Vector.empty,
+                                Map(localAddress.get -> Bitcoins(4),
+                                    remoteAddress.get -> Bitcoins(4)))
+        transactionResult <- client.fundRawTransaction(transactionWithoutFunds)
```

Comment:
So is the thing blocking you from using the bitcoin-s wallet is the fact we don't have a `.fundRawTransaction()`/`signRawTransactionWithWallet()`?

Reply:
It also doesn't have any functionality to broadcast to RegTest, in short I don't see how it has any utility here.

Reply:
Network functionality should not be touching signing functionality. They are two different systems that should be kept far apart for security reasons. I'm interested in what is blocking from signing be executed with the bitcoin-s wallet, is it the two methods I specified above?

Reply:
I believe it is just `fundRawTransaction` since that creates inputs (with signatures present). Also, do note that in this iteration Bitcoind RPC client is only used in test and the place where you are proposing I use bitcoin-s wallet is in that test. Nowhere am I using bitcoind for signing functionality in actual source code, only as a setup to a test, which I think should be acceptable?

Reply:
Well technically the way you have it setup the `testkit` is in `src` in the build.sbt file which brings in the entire testkit to the classpath in source. 

I think this is fine for now with #896 filed. We have an understanding of moving away from the DLC project in the future to having things in the wallet and core project.